Weather.py

```diff
@@ -0,0 +1,281 @@
+        if extracted_tuple.MinTemperatureC != "" and min_temp > \
+                int(extracted_tuple.MinTemperatureC):
+            min_temp = int(extracted_tuple.MinTemperatureC)
+            min_temp_day = resolve_date(extracted_tuple.PKT)
+
+        # finding maximum humidity
+        if extracted_tuple.MaxHumidity != "" and max_humid < \
+                int(extracted_tuple.MaxHumidity):
+            max_humid = int(extracted_tuple.MaxHumidity)
+            max_humid_day = resolve_date(extracted_tuple.PKT)
+
+    result = \
+        [
+            max_temp, max_temp_day,
+            min_temp, min_temp_day,
+            max_humid, max_humid_day
+        ]
+
+    return result
+
+
+''' task 02 to extract values of highest average temperature, lowest average
+temperature and average humidity in a month '''
+
+
+def monthly_average_temperature_and_humidity(data):
+
+    high_avg_temp = 0
+    high_avg_temp_count = 0
+    low_avg_temp = 0
+    low_avg_temp_count = 0
+    mean_avg_humid = 0
+    mean_avg_humid_count = 0
+
+    for extracted_tuple in data:
+
+        # finding highest average temperature
+        if extracted_tuple.MaxTemperatureC != "":
+            high_avg_temp += int(extracted_tuple.MaxTemperatureC)
+            high_avg_temp_count += 1
+
+        # finding lowest average temperature
+        if extracted_tuple.MinTemperatureC != "":
+            low_avg_temp += int(extracted_tuple.MinTemperatureC)
+            low_avg_temp_count += 1
+
+        # finding mean average humidity
+        if extracted_tuple.MeanHumidity != "":
+            mean_avg_humid += int(extracted_tuple.MeanHumidity)
+            mean_avg_humid_count += 1
+
+    result = \
+        [
+            int(high_avg_temp / high_avg_temp_count),
+            int(low_avg_temp / low_avg_temp_count),
+            int(mean_avg_humid / mean_avg_humid_count)
+        ]
+
+    return result
+
+
+''' task 03 to visualize a chart for highest temperature and lowest
+temperature of each day in a month '''
+
+
+def monthly_chart_temperature_and_humidity(data):
+
+    chart = dict()
+    chart = {"max_temp": [], "min_temp": []}
+
+    for extracted_tuple in data:
+
+        # finding highest average temperature
+        if extracted_tuple.MaxTemperatureC != "":
+            chart["max_temp"].append(extracted_tuple.MaxTemperatureC)
+
+        # finding lowest average temperature
+        if extracted_tuple.MinTemperatureC != "":
+            chart["min_temp"].append(extracted_tuple.MinTemperatureC)
+
+    return chart
+
+
+''' a utility function for mapping month names against the month number '''
+
+
+def mapping_months(month_number):
+
+    month_name = \
+        [
+            "Jan",
+            "Feb",
+            "Mar",
+            "Apr",
+            "May",
+            "Jun",
+            "Jul",
+            "Aug",
+            "Sep",
+            "Oct",
+            "Nov",
+            "Dec"
+        ]
+
+    return month_name[month_number - 1]
```

Comment:
better use dict globally. It will be much faster. Also investigate reason why dict are faster then array/list?